nixos/modules/services/networking/prosody.nix

```diff
@@ -1,885 +1,659 @@
+    (mkRenamedOptionModule [ "services" "prosody" "admins" ] [ "services" "prosody" "settings" "admins" ])
+    (mkRenamedOptionModule [ "services" "prosody" "authentication" ] [ "services" "prosody" "settings" "authentication" ])
+    (mkChangedOptionModule [ "services" "prosody" "disco_items" ] [ "services" "prosody" "settings" "disco_items" ] (config:
+      map ({ url, description }: [ url description ]) config.services.prosody.disco_items
+    ))
+    (mkChangedOptionModule [ "services" "prosody" "uploadHttp" ] [ "services" "prosody" "components" ] (config:
+      let
+        cfg = config.services.prosody;
+      in
+      {
+        ${cfg.uploadHttp.domain} = {
+          module = "http_upload";
+          # these values are to preserve compatibility with this module pre nixos 23.11
+          settings = {
+            http_upload_file_size_limit = cfg.uploadHttp.uploadFileSizeLimit or "50 * 1024 * 1024";
+            http_upload_expire_after = cfg.uploadHttp.uploadExpireAfter or "60 * 60 * 24 * 7";
+            http_upload_path = cfg.uploadHttp.httpUploadPath or "/var/lib/prosody";
+          } // optionalAttrs (cfg.uploadHttp ? userQuota) {
+            http_upload_quota = cfg.uploadHttp.userQuota;
+          };
+        };
+      }
+    ))
+    (mkChangedOptionModule [ "services" "prosody" "muc" ] [ "services" "prosody" "components" ] (config:
+      listToAttrs (map (muc: {
+        name = muc.domain;
+        value = {
+          module = "muc";
+          extraConfig = muc.extraConfig or "";
+          # these values are to preserve compatibility with this module pre nixos 23.11
+          settings = {
+            modules_enabled = optionals (muc.vcard_muc or true) [ "vcard_muc" ];
+            name = muc.name or "Prosody Chatrooms";
+            restrict_room_creation = muc.restrictRoomCreation or false;
+            max_history_messages = muc.maxHistoryMessages or 20;
+            muc_room_locking = muc.roomLocking or true;
+            muc_room_lock_timeout = muc.roomLockTimeout or 300;
+            muc_tombstones = muc.tombstones or true;
+            muc_tombstone_expiry = muc.tombstoneExpiry or 2678400;
+            muc_room_default_public = muc.roomDefaultPublic or true;
+            muc_room_default_members_only = muc.roomDefaultMembersOnly or false;
+            muc_room_default_moderated = muc.roomDefaultModerated or false;
+            muc_room_default_public_jids = muc.roomDefaultPublicJids or false;
+            muc_room_default_change_subject = muc.roomDefaultChangeSubject or false;
+            muc_room_default_history_length = muc.roomDefaultHistoryLength or 20;
+            muc_room_default_language = muc.roomDefaultLanguage or "en";
+          };
+        };
+      }) config.services.prosody.muc)
+    ))
+  ];
 
-    http_files = mkOption {
-      type = types.bool;
-      default = true;
-      description = lib.mdDoc "Serve static files from a directory over HTTP";
-    };
+  # interface
```

Comment:
```suggestion
```

Reply:
i thought that was a nice convention for new people but yeah... whatever 🤷‍♂️